LoopKitUI/Views/Settings Editors/TherapySettingsViewModel.swift

```diff
@@ -79,9 +95,10 @@ public class TherapySettingsViewModel: ObservableObject {
         therapySettings.workoutTargetRange = workout?.doubleRange(for: unit)
         didSave?(TherapySetting.workoutCorrectionRangeOverride, therapySettings)
     }
-    
-    public func saveSuspendThreshold(value: GlucoseThreshold) {
-        therapySettings.suspendThreshold = value
+
+    public func saveSuspendThreshold(quantity: HKQuantity) {
+        let settingsGlucoseUnit = therapySettings.glucoseUnit ?? preferredGlucoseUnit
+        therapySettings.suspendThreshold = GlucoseThreshold(unit: settingsGlucoseUnit, value: quantity.doubleValue(for: settingsGlucoseUnit))
```

Comment:
[nit] I'm a bit puzzled by this change here.  Not blocking, but care to explain?

Reply:
Sure. On the basic level, this logic was in the view, which seemed inappropriate. The view was already calling the `viewModel` to save the value, so it made sense to move that logic here.

The unit selection is trying to keep the original units.

Let me know if you want additional explanation.